### PR TITLE
Add PIO support to rp2350

### PR DESCRIPTION
**Pull request by Grazfather**

### CODE REVIEW

File: port/raspberrypi/rp2xxx/src/hal/pio.zig [hunk 1]
```diff
@@ -3,644 +3,92 @@ const std = @import("std");
...
-    };
-};
-
-pub fn PinMapping(comptime Count: type) type {
-    return struct {
-        base: u5 = 0,
-        count: Count = 0,
-    };
+pub const ShiftOptions = chip_specific.ShiftOptions;
+// TODO: THis works but we probably should use the chip_specific stuff
```
Comment (Grazfather) [inline port/raspberrypi/rp2xxx/src/hal/pio.zig:34]: Not sure if you'd rather put _everything_ unique in its own file, or use this stuff when able.

File: examples/raspberrypi/rp2xxx/src/ws2812.zig [hunk 2]
```diff
@@ -32,12 +32,18 @@ const ws2812_program = blk: {
...
+const led_pin = gpio.num(22);
+const power_pin = gpio.num(23);
 
 pub fn main() void {
     pio.gpio_init(led_pin);
+    // seeed xiao needs power applied to the ws2812 led
+    power_pin.set_function(.sio);
+    power_pin.set_direction(.out);
+    power_pin.put(1);
+
```
Comment (Grazfather) [inline examples/raspberrypi/rp2xxx/src/ws2812.zig:46]: Will remove this stuff. Needed for testing with the hardware I have

File: port/raspberrypi/rp2xxx/build.zig [hunk 3]
```diff
@@ -45,7 +45,7 @@ pub fn init(dep: *std.Build.Dependency) Self {
                 .{ .kind = .flash, .offset = 0x10000000, .length = 256 },
                 .{ .kind = .ram, .offset = 0x20000000, .length = 256 * 1024 },
             },
-            .patches = @import("patches/rp2040.zig").patches,
+            // .patches = @import("patches/rp2040.zig").patches,
```
Comment (Grazfather) [inline port/raspberrypi/rp2xxx/build.zig:48]: Patches don't apply to the new SVD

File: port/raspberrypi/rp2xxx/src/hal/pio/rp2040.zig [hunk 4]
```diff
@@ -0,0 +1,112 @@
...
+
+const microzig = @import("microzig");
+
+const common = @import("common.zig");
+
+const gpio = @import("../gpio.zig");
+const resets = @import("../resets.zig");
+const hw = @import("../hw.zig");
+
+pub const Pio = enum(u2) {
```
Comment (Grazfather) [inline port/raspberrypi/rp2xxx/src/hal/pio/rp2040.zig:12]: oops should be u1

File: port/raspberrypi/rp2xxx/src/hal/pio.zig [hunk 5]
```diff
@@ -3,687 +3,49 @@ const std = @import("std");
...
 
-const gpio = @import("gpio.zig");
-const resets = @import("resets.zig");
-const hw = @import("hw.zig");
 pub const assembler = @import("pio/assembler.zig");
 const encoder = @import("pio/assembler/encoder.zig");
-const cpu = @import("compatibility.zig").cpu;
 
 // global state for keeping track of used things
 var used_instruction_space: [2]u32 = [_]u32{ 0, 0 };
```
Comment (Grazfather) [inline port/raspberrypi/rp2xxx/src/hal/pio.zig:27]: probably should use the one in common!

Comment (Grazfather) [reply port/raspberrypi/rp2xxx/src/hal/pio.zig:27]: actually just remove

File: port/raspberrypi/rp2xxx/src/hal/pio/rp2040.zig [hunk 6]
```diff
@@ -0,0 +1,112 @@
...
+
+const gpio = @import("../gpio.zig");
+const resets = @import("../resets.zig");
+const hw = @import("../hw.zig");
+
+pub const Pio = enum(u1) {
+    pio0 = 0,
+    pio1 = 1,
+
+    pub fn reset(self: Pio) void {
```
Comment (haydenridd) [inline port/raspberrypi/rp2xxx/src/hal/pio/rp2040.zig:16]: Not a bug you introduced but go ahead and delete this function for now. It won't compile as is if you actually tried to use this function anways. For context this is the _hard_ reset bit for a given peripheral at the processor level and generally there's no real reason you want to do this. If someone really wants to do it they can use the `resets.zig` module! Will comment on the rp2350 implementation but same deal.

File: port/raspberrypi/rp2xxx/src/hal/pio/rp2350.zig [hunk 7]
```diff
@@ -0,0 +1,125 @@
...
+const hw = @import("../hw.zig");
+
+const PIO2 = microzig.chip.peripherals.PIO2;
+
+pub const Pio = enum(u2) {
+    pio0 = 0,
+    pio1 = 1,
+    pio2 = 2,
+
+    pub fn reset(self: Pio) void {
```
Comment (haydenridd) [inline port/raspberrypi/rp2xxx/src/hal/pio/rp2350.zig:19]: Same deal as the other one, go ahead and delete this method for now

File: examples/raspberrypi/rp2xxx/build.zig [hunk 8]
```diff
@@ -15,18 +15,16 @@ pub fn build(b: *std.Build) void {
...
         .{ .target = mb.ports.rp2xxx.boards.raspberrypi.pico, .name = "pico_spi-host", .file = "src/rp2040_only/spi_host.zig" },
-        .{ .target = mb.ports.rp2xxx.boards.raspberrypi.pico, .name = "pico_squarewave", .file = "src/rp2040_only/squarewave.zig" },
         .{ .target = mb.ports.rp2xxx.boards.raspberrypi.pico, .name = "pico_uart-echo", .file = "src/rp2040_only/uart_echo.zig" },
         .{ .target = mb.ports.rp2xxx.boards.raspberrypi.pico, .name = "pico_uart-log", .file = "src/rp2040_only/uart_log.zig" },
-        .{ .target = mb.ports.rp2xxx.boards.raspberrypi.pico, .name = "pico_usb-hid", .file = "src/rp2040_only/usb_hid.zig" },
-        .{ .target = mb.ports.rp2xxx.boards.raspberrypi.pico, .name = "pico_usb-cdc", .file = "src/rp2040_only/usb_cdc.zig" },
-        .{ .target = mb.ports.rp2xxx.boards.raspberrypi.pico, .name = "pico_ws2812", .file = "src/rp2040_only/ws2812.zig" },
-        .{ .target = mb.ports.rp2xxx.boards.raspberrypi.pico, .name = "pico_multicore", .file = "src/rp2040_only/blinky_core1.zig" },
+        // .{ .target = mb.ports.rp2xxx.boards.raspberrypi.pico, .name = "pico_usb-hid", .file = "src/rp2040_only/usb_hid.zig" },
```
Comment (haydenridd) [inline examples/raspberrypi/rp2xxx/build.zig:25]: Are these commented out examples intentional?

**Review by haydenridd — CHANGES_REQUESTED** [review]
Looking good! No complaints on my end, just delete that old method I spied when reviewing and need some clarification on some of the example stuff.

File: examples/raspberrypi/rp2xxx/build.zig (hunk 8 above)
Comment (Grazfather) [reply examples/raspberrypi/rp2xxx/build.zig:25]: Oops, no they were commented out because I needed #264 for them to work.

**Review by haydenridd — APPROVED** [review]
Looks good to me if you're ready!

**Pull request merged**